cmd/gen-go-sample/main.go

```diff
@@ -98,34 +98,45 @@ func main() {
 }
 
 func genMethodSamples(gen *generator, sampConf schema_v1p2.SampleConfig, nofmt bool, outDir string) error {
+	
 	for _, samp := range sampConf.Samples {
-		for _, iface := range gen.gapic.Interfaces {
-			if iface.Name != samp.Service {
-				continue
-			}
-			for _, meth := range iface.Methods {
-				if meth.Name != samp.Rpc {
-					continue
-				}
-				gen.reset()
-				if err := gen.genSample(samp, meth); err != nil {
-
-					err = errors.E(err, "generating: %s:%s", iface.Name+"."+meth.Name, samp.ID)
-					log.Fatal(err)
-				}
-
-				content, err := gen.commit(!nofmt, time.Now().Year())
-				if err != nil {
-					return err
-				}
-				// Handle duplicate sample IDs
-				fname := samp.ID + ".go"
-				if err := ioutil.WriteFile(filepath.Join(outDir, fname), content, 0644); err != nil {
-					return err
+		var iface GAPICInterface
+		var method GAPICMethod
+		for _, i := range gen.gapic.Interfaces {
+			if i.Name == samp.Service {
+				iface = i
+				for _, m := range iface.Methods {
+					if m.Name != samp.Rpc {
+						method = m
+						break
+					}
 				}
 				break
 			}
-			break
+		}
+
+		if method.Name == "" {
+			return errors.E(nil, "generating sample %q: rpc %q not found", samp.ID, method.Name)
+		}
+		if iface.Name == "" {
+			return errors.E(nil, "generating sample %q: service %q not found", samp.ID, iface.Name)
+		}
+
+		gen.reset()
+		if err := gen.genSample(samp, method); err != nil {
+
+			err = errors.E(err, "generating: %s:%s", iface.Name+"."+method.Name, samp.ID)
```

Comment:
Would be simpler to just do `%s.%s:%s` in the format string and omit string concat.

Reply:
Done.